python/cudf_polars/cudf_polars/experimental/benchmarks/utils.py

```diff
@@ -144,14 +145,32 @@ def from_args(cls, args: argparse.Namespace) -> RunConfig:
         if executor == "in-memory" or executor == "cpu":
             scheduler = None
 
+        path = args.path
+        if (scale_factor := args.scale) is None:
+            if path is None:
+                raise ValueError(
+                    "Must specify --root and --scale if --path is not specified."
+                )
+            # Use supplier table to estimate scale-factor
+            supplier = get_data(path, "supplier", args.suffix)
+            num_rows = supplier.select(pl.len()).collect().item(0, 0)
+            scale_factor = num_rows / 10_000
+        if path is None:
+            path = f"{args.root}/scale-{scale_factor}"
+        try:
+            scale_factor = int(scale_factor)
+        except ValueError:
+            scale_factor = float(scale_factor)
+
```

Comment:
*suggestion*: If we're given a scale factor, shall we check the data matches what we were told?